### PR TITLE
Persistent Mutes - For when you really don't want them spamming prayers

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Adds a persistent mute system for admins to take advantage of, and adds a table into the database to store those mutes.
It tells the client, if they are connected, what they were muted from and why.
Supports multiple mute reasons!
## Why It's Good For The Game
@Timberpoes and I were discussing the fact that mutes should really be persistent.
@Mothblocks wanted them to have a reason for tracking purposes if made persistent.

image
image
image
image
image
image
image
## Changelog
:cl:
admin: Mutes are now persistent and applied/revoked using the associated verb.
server: Database Schema updated. Read the changelog!
/:cl:

### CODE REVIEW

File: config/dbconfig.txt
Comment (Mothblocks) [inline config/dbconfig.txt:1]: You need to revert all these changes

File: code/modules/admin/verbs/admingame.dm [hunk 1]
```diff
@@ -78,15 +78,6 @@
...
-		body += "<br><b>Mute: </b> "
-		body += "\[<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_IC]'><font color='[(muted & MUTE_IC)?"red":"blue"]'>IC</font></a> | "
-		body += "<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_OOC]'><font color='[(muted & MUTE_OOC)?"red":"blue"]'>OOC</font></a> | "
-		body += "<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_PRAY]'><font color='[(muted & MUTE_PRAY)?"red":"blue"]'>PRAY</font></a> | "
-		body += "<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_ADMINHELP]'><font color='[(muted & MUTE_ADMINHELP)?"red":"blue"]'>ADMINHELP</font></a> | "
-		body += "<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_INTERNET_REQUEST]'><font color='[(muted & MUTE_INTERNET_REQUEST)?"red":"blue"]'>WEBREQ</font></a> | "
-		body += "<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_DEADCHAT]'><font color='[(muted & MUTE_DEADCHAT)?"red":"blue"]'>DEADCHAT</font></a>\]"
-		body += "(<A href='?_src_=holder;[HrefToken()];mute=[M.ckey];mute_type=[MUTE_ALL]'><font color='[(muted & MUTE_ALL)?"red":"blue"]'>toggle all</font></a>)"
```
Comment (Mothblocks) [inline code/modules/admin/verbs/admingame.dm:89]: Woah, this is REALLY not good. The player panel is the sanest places for these all to be

**Comment by Mothblocks** [comment]
i think that this is *extremely* overkill and worse than the existing ban system. for example, you can no longer check your admin remarks in order to see everything you can appeal. 

you can *already* ban people from praying, OOCing, etc. why do we need *another* system for this? i would close this outright but @MrStonedOne does not really want me touching admin stuff so i'm instead going to poke them for a sanity check

**Comment by ZephyrTFA** [comment]
Sad

File: code/modules/admin/sql_mute_system.dm [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+				mute_information.id,
+				params["muted_flag"],
+				params["reason"],
+				)
+			if(!edit_successful)
+				to_chat(usr, span_alertwarning("Failed to edit persistent mute."))
+				return TRUE
+
+			to_chat(usr, span_adminnotice("Edited persistent mute."))
+			ASYNC
```
Comment (LemonInTheDark) [inline code/modules/admin/sql_mute_system.dm:204]: I hate you so much

**Comment by Jordie0608** [comment]
For reference, these are the previous 3 times something along these lines was rejected:
#61875 #27172 #24268

**Pull request closed without merging**